Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -314,6 +318,54 @@ void CookieStore::remove(CookieStoreDeleteOptions&& options, Ref<DeferredPromise
     set(WTFMove(initOptions), WTFMove(promise));
 }
 
+void CookieStore::cookiesAdded(const String& host, const Vector<Cookie>& cookies)
+{
+    if (!m_hasChangeEventListener)
+        return;
+
+    auto* context = scriptExecutionContext();
+    if (!context)
+        return;
+
+    auto& document = *downcast<Document>(context);
+    if (host != document.url().host().toString())
+        return;
+
+    Vector<CookieListItem> cookieList;
+    for (auto cookie : cookies)
+        cookieList.append(CookieListItem { cookie });
```

Comment:
I believe you'll like need to go though the cookies to see if they have an expiration in the past. If they do, add them to the deleted list instead of the changed list.

Reply:
Yeah, this makes sense. But what do we do with `cookiesDeleted` since it isn't being called at all?

Reply:
It gets called when specific CFNetwork deletion API gets used so you still need to handle it.